File: CSharp/src/MagnetarYggdrasil/Models/Task.cs
namespace MagnetarYggdrasil.Models;

public class TaskItem
{
    public int Id { get; set; }
    public int UserId { get; set; }
    public required string Title { get; set; }
    public required string Description { get; set; }
    public bool IsDone { get; set; }
}

File: Rust/src/models/task.rs
use serde::{Deserialize, Serialize};
use sqlx::FromRow;

#[derive(Debug, Serialize, Deserialize, FromRow, Clone)]
pub struct Task {
    pub id: i64,
    pub user_id: i64,
    pub title: String,
    pub description: String,
    pub is_done: bool,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct CreateTask {
    pub title: String,
    pub description: String,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct UpdateTask {
    pub title: Option<String>,
    pub description: Option<String>,
    pub is_done: Option<bool>,
}
